### DISCUSSION TITLE
Reuse Hook

### DISCUSSION THREAD

**Discussion by achempak-polymer** [aarondl/sqlboiler#1206] [Q&A] [answered]
Is there a convenient way to write a hook such that it can be reused by all models? For example, say I have a `created_by` column in all of my tables, and I want this value to be set whenever a row is created in a table. Do I need to write a separate hook for each model, or is there some way I can write a single hook and apply it to every model?

**Comment by stephenafamo** [comment, marked as answer]
First, for `created_at` columns, SQLBoiler can add automatic timestamps, so nothing to do on your end.

But to answer your question about re-using hooks. There is no easy way to do that.

You can consider extending the templates with the `--templates` flag to generate code for whatever new functionality you want to add, but that is not the most intuitive process.
